### PR TITLE
Add `loss` attribute for backward compatibility

### PR DESCRIPTION
**Pull request by haifeng-jin**
The old implementation is here: https://github.com/keras-team/keras/blob/v2.14.0-rc0/keras/engine/training.py#L823

### CODE REVIEW

File: keras_core/trainers/trainer.py [hunk 1]
```diff
@@ -18,6 +18,7 @@ def __init__(self):
         self._run_eagerly = False
         self._jit_compile = None
         self.compiled = False
+        self.loss = {}
```
Comment (fchollet) [inline keras_core/trainers/trainer.py:21]: Should be `None` no?

Comment (haifeng-jin) [reply keras_core/trainers/trainer.py:21]: In `tf.keras`, it is `{}`.
https://colab.research.google.com/drive/1DEr6w72wwuNGR9MqNlB_sgTenSBdLBge?resourcekey=0-4Yj0umES4aNjX2s0Qc6gTA&usp=sharing

Would you like it to be `None`?
I am OK with either one.

Comment (fchollet) [reply keras_core/trainers/trainer.py:21]: `None` is the signature default so it's much more sensible than an empty dicts (`loss` can be a dict but it's not a common case).

Comment (haifeng-jin) [reply keras_core/trainers/trainer.py]: Changed to `None`.

**Review by fchollet — APPROVED** [review]
LGTM, thanks
